src/runmode-dpdk.c

```diff
@@ -0,0 +1,1274 @@
+                    .offloads = 0,
+            },
+    };
+
+    // configure RX offloads
+    if (dev_info->rx_offload_capa & DEV_RX_OFFLOAD_RSS_HASH) {
+        if (iconf->nb_rx_queues > 1) {
+            SCLogConfig("RSS enabled on %s for %d queues", iconf->iface, iconf->nb_rx_queues);
+            port_conf->rx_adv_conf.rss_conf.rss_key = rss_hkey;
+            port_conf->rx_adv_conf.rss_conf.rss_key_len = RSS_HKEY_LEN;
+            port_conf->rx_adv_conf.rss_conf.rss_hf = ETH_RSS_TCP | ETH_RSS_UDP;
+            port_conf->rxmode.mq_mode = ETH_MQ_RX_RSS;
+        } else {
+            SCLogConfig("RSS not enabled on %s", iconf->iface);
+            port_conf->rx_adv_conf.rss_conf.rss_key = NULL;
+            port_conf->rx_adv_conf.rss_conf.rss_hf = 0;
+        }
+    } else {
+        SCLogConfig("RSS not supported on %s", iconf->iface);
+    }
+
+    if (iconf->checksum_mode == CHECKSUM_VALIDATION_DISABLE) {
+        SCLogConfig("Checksum validation disabled on %s", iconf->iface);
+    } else if (dev_info->rx_offload_capa & DEV_RX_OFFLOAD_CHECKSUM) {
+        if (iconf->checksum_mode == CHECKSUM_VALIDATION_ENABLE &&
+                iconf->flags & DPDK_RX_CHECKSUM_OFFLOAD) {
+            SCLogConfig("Checksum validation enabled and offloaded on %s", iconf->iface);
+            port_conf->rxmode.offloads |= DEV_RX_OFFLOAD_CHECKSUM;
+        } else if (iconf->checksum_mode == CHECKSUM_VALIDATION_ENABLE &&
+                   !(iconf->flags & DPDK_RX_CHECKSUM_OFFLOAD)) {
+            SCLogConfig("Checksum validation enabled but not offloaded on %s", iconf->iface);
+        }
+    }
+
+    // configure TX offloads
+    if (dev_info->tx_offload_capa & DEV_TX_OFFLOAD_MT_LOCKFREE) {
+        port_conf->txmode.offloads |= DEV_TX_OFFLOAD_MT_LOCKFREE;
+    }
+    if (dev_info->tx_offload_capa & DEV_TX_OFFLOAD_MBUF_FAST_FREE) {
+        port_conf->txmode.offloads |= DEV_TX_OFFLOAD_MBUF_FAST_FREE;
+    }
+
+    port_conf->fdir_conf = fdir_conf;
+}
+
+int DeviceConfigureQueues(DPDKIfaceConfig *iconf, const struct rte_eth_dev_info *dev_info,
+        const struct rte_eth_conf *port_conf)
+{
+    int retval;
+    struct rte_eth_rxconf rxq_conf;
+    struct rte_eth_txconf txq_conf;
+
+    iconf->pkt_mempool_array = SCCalloc(iconf->nb_rx_queues, sizeof(struct rte_mempool *));
+    if (unlikely(iconf->pkt_mempool_array == NULL)) {
+        SCLogError(SC_ERR_MEM_ALLOC, "Could not allocate memory for packet mempool pointers");
+        SCReturnInt(-ENOMEM);
+    }
+
+    for (uint16_t queue_id = 0; queue_id < iconf->nb_rx_queues; queue_id++) {
+        char mempool_name[64];
+        snprintf(mempool_name, 64, "pktmbuf_pool_p%d_q%d", iconf->port_id, queue_id);
+        SCLogInfo("Creating a packet mbuf pool %s of size %d, cache size %d, mbuf size %d",
+                mempool_name, iconf->mempool_size, iconf->mempool_cache_size,
+                RTE_MBUF_DEFAULT_BUF_SIZE);
+
+        iconf->pkt_mempool_array[queue_id] = rte_pktmbuf_pool_create(mempool_name,
```

Comment:
if the intention is to create mempool per NUMA, the arguments passed for DPDK `pool_create` is incorrect. If the intention was not to create mempool per NUMA, why one needs to create mempool multiple times when single mempool can be used for all NIC. Also caching done per core is much better.

Reply:
thanks, please also add warning if copy-interface is different NUMA stating the performance will differ.